components/i2cdev/i2cdev.h

```diff
@@ -68,12 +73,23 @@ extern "C" {
 typedef struct
 {
     i2c_port_t port;         //!< I2C port number
-    i2c_config_t cfg;        //!< I2C driver configuration
-    uint8_t addr;            //!< Unshifted address
-    SemaphoreHandle_t mutex; //!< Device mutex
+#if defined(CONFIG_I2CDEV_USING_LEGACY_I2C)
+    i2c_config_t cfg;        //!< I2C driver configuration (i2c.h only)
     uint32_t timeout_ticks;  /*!< HW I2C bus timeout (stretch time), in ticks. 80MHz APB clock
                                   ticks for ESP-IDF, CPU ticks for ESP8266.
-                                  When this value is 0, I2CDEV_MAX_STRETCH_TIME will be used */
+                                  When this value is 0, I2CDEV_MAX_STRETCH_TIME will be used
+                                  (i2c.h only) */
+#else
+    i2c_master_bus_handle_t bus; //!< I2C master bus handle (i2c_master.h only)
+    i2c_master_dev_handle_t device; //!< I2C master bus device handle
+                                    //(i2c_master.h only)
+    i2c_device_config_t device_config; //!< I2C device configuration (i2c_master.h only)
+    i2c_master_bus_config_t master_bus_config; /*! < I2C master bus specific
```

Comment:
It understand that the i2cdev component could stay in charge of the i2c bus, such as it was when using the previous I2C driver. Espressif added the option for queuing and thread safety, allowing multiple components/libraries to share the bus. Taking ownership of a bus to (again) be used by only this component creates the same conflicts as with the previous version: needing to either use a different bus for a device that is not supported by esp-idf-lib or (now that a bus can be shared) "digging" into this (i2cdev) component for another component to be able to share the bus. In my (very junior) opinion, would it not provide more flexibility to let the develop instantiate the bus (as you already now in the main.c file) and pass it as a pointer, for i2cdev to access (too)?